server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/Email.scala

```diff
@@ -446,42 +447,47 @@ private class EmailFullViewFactory @Inject()(zoneIdProvider: ZoneIdProvider, pre
 }
 
 private class EmailMetadataViewReader @Inject()(messageIdManager: MessageIdManager,
-                                                metadataViewFactory: EmailMetadataViewFactory) extends EmailViewReader {
-  private val reader: GenericEmailViewReader = new GenericEmailViewReader(messageIdManager, MINIMAL, metadataViewFactory)
+                                                metadataViewFactory: EmailMetadataViewFactory) extends EmailViewReader[EmailMetadataView] {
+  private val reader: GenericEmailViewReader[EmailMetadataView] = new GenericEmailViewReader[EmailMetadataView](messageIdManager, MINIMAL, metadataViewFactory)
 
-  override def read(ids: Seq[MessageId], request: EmailGetRequest, mailboxSession: MailboxSession): SFlux[EmailView] =
+  override def read[T >: EmailMetadataView](ids: Seq[MessageId], request: EmailGetRequest, mailboxSession: MailboxSession): SFlux[T] =
```

Comment:
are you sure you need a parameterized type here ?

Reply:
Yes, overwise it do not compile